### DISCUSSION TITLE
Exit Meilisearch after dump is imported

### DISCUSSION THREAD

**Discussion by WilliamDEdwards** [meilisearch/product#792] [Feedback & Feature Proposal]
Situation:

- Meilisearch is run under systemd
- Meilisearch is upgraded, so the dump has to be imported with `meilisearch --import-dump`

Problem: when the dump is imported, Meilisearch runs in the foreground. Instead, Meilisearch should quit immediately after importing the dump is done, so it can be started properly under systemd.

Otherwise, `--import-dump` has to be temporarily passed to the systemd service, then somehow one has to detect that the dump was imported, then the systemd service can be restarted without the argument. This is unnecessarily ugly and complex.

Admittedly, it is possible to always pass `--import-dump` and ignore that when a database exists using `--ignore-dump-if-db-exists`, but that's of the limited usefulness due to the instability of the dump path created by `POST /dumps`.

**Comment by macraig** [comment]
Hi @WilliamDEdwards , thanks for sharing your feedback about the upgrade process. We agree that the current process is far from ideal, so you're not alone in wanting it to be simpler 😄 

The good news is that we’re already working on a "Dumpless Upgrades" solution that will make version upgrades seamless without needing manual dump imports. This is planned for release in 2025. You can check out more about this on the public [page](https://www.notion.so/meilisearch/Dumpless-upgrade-fff4b06b651f81f1acafe24d4687b3f7) (it's a work in progress, so it might change as we work on the feature)

Thanks for sticking with us while we improve Meilisearch!

**Reply by WilliamDEdwards** [reply]
Thanks, @macraig. Good to know the upgrade process will be simplified. 

P.S. Please keep your profile picture forever ;)
